File: 1.Data structure/2.Queue/기능 개발/기능 개발.py
def solution(progresses, speeds):
    answer = []
    cnt = 1
    for i in range(0,len(progresses)):
        speeds[i] = -((progresses[i]-100)//speeds[i])
        if i >= 1 and max(speeds[:i]) >= speeds[i]:
            cnt+=1
        elif i != 0:
            answer.append(cnt)
            cnt = 1
    answer.append(cnt)
    return answer



# def solution(progresses, speeds):
#     answer = [0]
#     index = 0
#     while min(progresses) < 100:
#         for i in range(index,len(speeds)):
#             progresses[i] += speeds[i]
#         while progresses[index] >= 100:
#             index += 1
#             if index >= len(speeds):
#                 break
#         answer.append(index)
#     answer = sorted(list(set(answer)))
#     return [answer[x+1]-answer[x] for x in range(0,len(answer)-1)]

File: 1.Data structure/2.Queue/기능 개발.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<Integer> solution(int[] progresses, int[] speeds) {
        int cnt = 1;
        int[] answer = new int[speeds.length];
        int iMax = 0;
        ArrayList<Integer> integers1 = new ArrayList<Integer>();
        for(int i = 0; i< progresses.length; i++){
            for(int j : answer){
                iMax = Math.max(j,iMax);
            }
            answer[i] = (int)Math.ceil((100-(double)progresses[i])/(double)speeds[i]);
            if (i>=1 && iMax >= answer[i]){
                cnt+=1;
            }else if(i != 0){
                integers1.add(cnt);
                cnt = 1;
            }
        }

        integers1.add(cnt);
        return integers1;
    }
}